dbt/scripts/export_models.py

```diff
@@ -0,0 +1,212 @@
+    dbt_output = io.StringIO()
+    with contextlib.redirect_stdout(dbt_output):
+        dbt_list_result = DBT.invoke(
+            [
+                "--quiet",
+                "list",
+                "--target",
+                target,
+                "--resource-types",
+                "model",
+                "--output",
+                "json",
+                "--output-keys",
+                "name",
+                "config",
+                "relation_name",
+                "--select",
+                *select,
+            ]
+        )
+
+    if not dbt_list_result.success:
+        print("Encountered error in `dbt list` call")
+        raise ValueError(dbt_list_result.exception)
+
+    # Output is formatted as a list of newline-separated JON objects
+    models = [
+        json.loads(model_dict_str)
+        for model_dict_str in dbt_output.getvalue().split("\n")
+        # Filter out empty strings caused by trailing newlines
+        if model_dict_str
+    ]
+
+    if not models:
+        raise ValueError(f"No models found for the --select value '{select}'")
+
+    print(
+        "The following models will be exported: "
+        f"{', '.join(model['name'] for model in models)}"
+    )
+
+    conn = pyathena.connect(
+        s3_staging_dir=os.getenv(
+            "AWS_ATHENA_S3_STAGING_DIR",
+            "s3://ccao-dbt-athena-results-us-east-1",
+        ),
+        region_name=os.getenv("AWS_ATHENA_REGION_NAME", "us-east-1"),
+    )
+
+    for model in models:
+        # Extract useful model metadata from the columns we queried in
+        # the `dbt list` call above
+        model_name = model["name"]
+        relation_name = model["relation_name"]
+        export_name = model["config"]["meta"].get("export_name") or model_name
+        template = (
+            model["config"]["meta"].get("export_template")
+            or f"{model_name}.xlsx"
+        )
+
+        # Define inputs and outputs for export based on model metadata
+        template_path = os.path.join("export", "templates", template)
+        template_exists = os.path.isfile(template_path)
+        output_path = os.path.join("export", "output", f"{export_name}.xlsx")
+
+        print(f"Querying data for model {model_name}")
+        model_df = pd.read_sql(f"SELECT * FROM {relation_name}", conn)
+
+        # Delete the output file if one already exists
+        pathlib.Path(output_path).unlink(missing_ok=True)
+
+        if template_exists:
+            print(f"Using template file at {template_path}")
+            shutil.copyfile(template_path, output_path)
+        else:
+            print("No template file exists; creating a workbook from scratch")
+
+        writer_kwargs = (
+            {"mode": "a", "if_sheet_exists": "overlay"}
+            if template_exists
+            else {}
+        )
+        with pd.ExcelWriter(
+            output_path, engine="openpyxl", **writer_kwargs
+        ) as writer:
+            sheet_name = "Sheet1"
```

Comment:
In the future we might want to consider supporting a config value to override sheet names, but for now it's easiest to use the default value.